Gems/PhysX/Code/Include/PhysX/UserDataTypes.h

```diff
@@ -86,7 +52,18 @@ namespace PhysX
 
         AzPhysics::SimulatedBody* GetSimulatedBody() const;
 
+        bool IsValid() const;
+
     private:
+        using PxActorUniquePtr = AZStd::unique_ptr<physx::PxActor, AZStd::function<void(physx::PxActor*)> >;
+
+        ///This is an arbitary value used to verify the cast from void* userdata pointer on a pxActor to ActorData
+        ///is safe. If m_sanity does not have this value, then it is not safe to use the casted pointer.
+        ///Helps to debug if someone is setting userData pointer to something other than this class during development
+        static const AZ::u32 s_sanityValue = 0xba5eba11;
```

Comment:
Not sure if it's needed for AZ::u32, but it'd be more correct to specify `inline` since the static member is initialized.

Reply:
It can actually be static constexpr. made it like that now